action/action.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	actions "github.com/sethvargo/go-githubactions"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+const defaultRepositoryOwner string = "hashicorp"
+const defaultMetadataFileName string = "metadata.json"
+
+type input struct {
+	branch           string
+	filePath         string
+	metadataFileName string
+	product          string
+	repository       string
+	repositoryOwner  string
+	sha              string
+	version          string
+}
+
+type Metadata struct {
+	Branch          string `json:"branch"`
+	BuildWorkflowId string `json:"buildWorkflowId"`
+	Product         string `json:"product"`
+	Repository      string `json:"repository""`
+	RepositoryOwner string `json:"repositoryOwner"`
+	Revision        string `json:"sha"`
+	Version         string `json:"version"`
+}
+
+func main() {
+	in := input{
+		branch:           actions.GetInput("branch"),
+		filePath:         actions.GetInput("filePath"),
+		metadataFileName: actions.GetInput("metadataFileName"),
```

Comment:
I see this is exactly what I wasn't sure about **metadataFileName** field currently defaults to the `metadata.json` filename if nothing is provided https://github.com/hashicorp/actions-generate-metadata/blob/action_files/action/action.go#L81-L83

The thinking behind this was if the action user wants to use a different filename (not metadata.json) in case this public action will be used by someone else 🙂 .

Reply:
That's fine!